mycroft/util/parse.py

```diff
@@ -164,6 +222,580 @@ def extractnumber_en(text):
     return val
 
 
+def extract_datetime_en(str, currentDate=None):
```

Comment:
I take a large part of the blame for this.  The original PHP function was also a beast, but I left it that way since the monster was at least caged in this one function.

I'd rather get this in sooner than later, so I asked him to come back l later to do a refactoring.